rest/admin_api_test.go

```diff
@@ -3749,3 +3749,86 @@ func TestLegacyCredentialInheritance(t *testing.T) {
 	)
 	assert.Equal(t, http.StatusCreated, resp.StatusCode)
 }
+
+func TestDatabaseOfflineConfigLegacy(t *testing.T) {
+	rt := NewRestTester(t, &RestTesterConfig{persistentConfig: false})
+	bucket := rt.Bucket()
+	defer rt.Close()
+
+	dbConfig := `{
+    "bucket": "` + bucket.GetName() + `",
+    "name": "db",
+    "sync": "function(doc){ channel(doc.channels); }",
+    "import_docs": true,
+    "offline": false,
+    "enable_shared_bucket_access": true,
+    "num_index_replicas": 0 }`
```

Comment:
You probably want to `use_views` here and have it read the `TestsDisableGSI`. Similarly we probably shouldn't always be setting `enable_shared_bucket_access` to true, have this read `TestUseXattrs`